### ISSUE TITLE
SMTP connection type clarifications are needed

### ISSUE THREAD

**Issue by johnjeffers** [apache/airflow#16468]
**Description**

The documentation on how to set up STMP is not clear.

https://airflow.apache.org/docs/apache-airflow/stable/howto/email-config.html

It says to create a connection named `smtp_default` but does not say what type of connection to create. There is no connection type named `SMTP`

It was suggested to me in Airflow Slack to create it as an `HTTP` connection, but this connection type does not contain all fields necessary to configure SMTP, particularly with TLS.

**Use case / motivation**

I would like the instructions for setting up SMTP in Airflow to be clearer, and it would make sense that there is an `SMTP` connection type with all necessary fields.

**Comment by msumit** [comment]
I think the correct section to setup SMTP for Airflow is here -> https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html?highlight=smtp#smtp

**Comment by potiuk** [comment]
Indeed when you want to try to configure SSL you should configure parameter there. You can also configure USER/PASSWORD this way, but we also have (apparently) a mechanism when you can additionally specify user/password in connection "by default it is 'smtp_default" - if that connection is available, login and password are taken from there. But no other values appaarently.

I think we need to clarify it and possibly make somewhat more consistent.

**Comment by potiuk** [comment]
> It was suggested to me in Airflow Slack to create it as an HTTP connection, but this connection type does not contain all fields necessary to configure SMTP, particularly with TLS.

So just to summarise: currently you can configurre SMTP email sending either entirely via configuration in '[smtp]' as pointed out by @msumit or you can additionally configure the user/password in the `smtp_default`. 

This is a bit complex and unintuitive and I think we need to at least explain it but possibly make it consistent.

**Comment by zachliu** [comment]
in case someone else is as confused as me...
#16523 is not included in 2.1.4 release, so https://airflow.apache.org/docs/apache-airflow/2.1.4/howto/email-config.html still gives me old info and this log entry
```
[2021-09-30 21:10:46,400] {{base.py:70}} INFO - Using connection to: id: smtp_default. Host: <my-ses-host>, Port: <my-port>, Schema: , Login: <my-user-name>, Password: ***, extra: {}
```
made me think i could put host/port/etc. into the connection until i carefully examined the [source code](https://github.com/apache/airflow/blob/614858fb7d443880451e6111b27fdaf942f563a4/airflow/utils/email.py#L184-L190) :crying_cat_face:
